sources/table.cpp

```diff
@@ -37,27 +30,61 @@ PositionList<int32_t>* Table::table_eq_scan(const int32_t &column_id, const int3
     return list;
 }
 
-
 std::random_device Table::randomDevice;
-std::mt19937 Table::randomGenerator = std::mt19937(Table::randomDevice()); // seed the generator
+std::mt19937 Table::randomGenerator = std::mt19937(Table::randomDevice());                                                                                                // seed the generator
 std::uniform_int_distribution<> Table::randomDistribution = std::uniform_int_distribution<>(std::numeric_limits<int32_t>::lowest(), std::numeric_limits<int32_t>::max()); // define the range
 
-int* Table::generateRandomDistinctValues(int32_t numberOfDistinctValues) {
-    int32_t* distinctValues = new int32_t[numberOfDistinctValues];
+int *Table::generateRandomDistinctValues(int32_t numberOfDistinctValues)
+{
+    int32_t *distinctValues = new int32_t[numberOfDistinctValues];
 
-    for (int i = 0; i < numberOfDistinctValues; i++) {
+    for (int i = 0; i < numberOfDistinctValues; i++)
+    {
         int32_t new_value = Table::randomDistribution(Table::randomGenerator);
         // TODO: should check if really distinct
         distinctValues[i] = new_value;
     }
     return distinctValues;
 }
 
-int* Table::generateDistinctValues(int32_t numberOfDistinctValues) {
-    int32_t* distinctValues = new int32_t[numberOfDistinctValues];
+int *Table::generateDistinctValues(int32_t numberOfDistinctValues)
+{
+    int32_t *distinctValues = new int32_t[numberOfDistinctValues];
 
-    for (int i = 0; i < numberOfDistinctValues; i++) {
+    for (int i = 0; i < numberOfDistinctValues; i++)
+    {
         distinctValues[i] = i;
     }
     return distinctValues;
 }
+
+void Table::generateData(int32_t rows, uint32_t *distinctValues)
+{
+    for (auto columnIndex = 0; columnIndex < m_columns; columnIndex++)
+    {
+
+        auto columnValues = Table::generateDistinctValues(distinctValues[columnIndex]);
+
+        for (auto rowIndex = 0; rowIndex < rows; rowIndex++)
+        {
+            auto valueIndex = rand() % distinctValues[columnIndex];
+            this->getLocation(rowIndex, columnIndex) = columnValues[valueIndex];
+        }
+        delete[] columnValues;
+    }
+    m_numRows += rows;
+}
+
+void Table::insert(int32_t *values)
+{
+    //check if there are enough values in input
+    assert(values[m_columns]);
+
+    if (m_numRows + 1 <= m_maxRows)

```

Comment:
nice for assert too, otherwise use bool and return false if it fails. handle else case otherwise! also a nice place for exceptions...